boa/src/builtins/json/mod.rs

```diff
@@ -65,10 +67,68 @@ pub fn parse(_: &mut Value, args: &[Value], _: &mut Interpreter) -> ResultValue
 ///
 /// [spec]: https://tc39.es/ecma262/#sec-json.stringify
 /// [mdn]: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/JSON/stringify
-pub fn stringify(_: &mut Value, args: &[Value], _: &mut Interpreter) -> ResultValue {
-    let obj = args.get(0).expect("cannot get argument for JSON.stringify");
-    let json = obj.to_json().to_string();
-    Ok(Value::from(json))
+pub fn stringify(_: &mut Value, args: &[Value], interpreter: &mut Interpreter) -> ResultValue {
+    let object = match args.get(0) {
+        Some(obj) => obj,
+        None => return Ok(Value::undefined()),
+    };
+
+    let replacer = match args.get(1) {
+        Some(replacer) if replacer.is_object() => replacer,
+        _ => return Ok(Value::from(object.to_json().to_string())),
+    };
+
+    let replacer_as_object = replacer
+        .as_object()
+        .expect("JSON.stringify replacer was an object");
+    if replacer_as_object.is_callable() {
+        object
+            .as_object()
+            .map(|obj| {
+                let object_to_return = Value::new_object(None);
+                for (key, val) in obj.properties.iter().filter_map(|(key, v)| {
+                    if let Some(value) = &v.value {
```

Comment:
We should use `Option::map()` here, that does exactly this.

Reply:
:heavy_check_mark: